src/app/file-panel.js

```diff
@@ -108,14 +112,16 @@ function filepanel (appAPI, files) {
 
   var events = new EventManager()
   var element = template()
-  // TODO please do not add custom javascript objects, which have no
-  // relation to the DOM to DOM nodes
-  element.events = events
+
   fileExplorer.events.register('focus', function (path) {
     appAPI.switchToFile(path)
   })
 
-  return element
+  var api = { events: events }
+
+  container.appendChild(element)
+
+  return api
```

Comment:
We should be consistent. If you use `new FooBar()`, then I would expect `FooBar` to be a constructor function that does not return explicitly. Please keep it simple, we do not need optimization here. My perception is that `new` without explicit return (and potentially prototypes but not necessarily) is the standard way. It is at least the standard way in this codebase.